tests/visco_plastic_derivatives_2d.cc

```diff
@@ -173,6 +173,7 @@ void f(const aspect::SimulatorAccess<dim> &simulator_access,
         {
           in_dviscositydstrainrate.strain_rate[i] = in_base.strain_rate[i]
                                                     + std::fabs(in_base.strain_rate[i][strain_rate_indices])
+                                                    * (component > dim-1 ? 0.5 : 1 )
                                                     * finite_difference_accuracy
                                                     * aspect::Utilities::nth_basis_for_symmetric_tensors<dim>(component);
```

Comment:
```suggestion
          // components that are not on the diagonal are multiplied by 0.5, because the symmetric tensor
          // is modified by 0.5 in both symmetric directions (xy/yx) simultaneously and we compute the combined
          // derivative
          in_dviscositydstrainrate.strain_rate[i] = in_base.strain_rate[i]
                                                    + std::fabs(in_base.strain_rate[i][strain_rate_indices])
                                                    * (component > dim-1 ? 0.5 : 1 )
                                                    * finite_difference_accuracy
                                                    * aspect::Utilities::nth_basis_for_symmetric_tensors<dim>(component);
```